text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,324 @@
+  in 6 months, and explain the policy that is applicable to newer releases.
+
+* **6 months later:** Formally announce that the `v1` release line of `jsii` is
+  entering the *Maintenance* tier of the [AWS SDKs and Tools maintenance
+  policy][aws-policy], and will continue to receive critical bug and security
+  fixes for 12 months before transitioning to *end-of-life*, and repeat the
+  maintenance policy that new releases will benefit from.
+
+* **12 months later:** Formally announce that the `v1` release line of `jsii` is
+  transitioning to *end-of-life* and will no longer receive any updates. Repeat
+  the maintenance policy that is applicable to newer releases.
+
+### Is this a breaking change?
+
+This is not a breaking change in the traditional sense, however this is a
+significant change in versioning schemes used for the `jsii` compiler, and a
+departure from the [AWS SDKs and Tools maintenance policy][aws-policy].
+
+As such, this warrants ample communication with our customer base to avoid they
+are caught by surprise, or have broken expectations.
+
+### What alternative solutions did you consider?
+
+#### Bring-your-own TypeScript version
+
+One of the main rationales for this change is to enable the use of new
+TypeScript releases without forcing the enitre package ecosystem to make the
+change at the same time (due to language-breaking changes).
+
+An option to address this would be to allow customers to "bring their own"
+TypeScript compiler, by making it a `peerDependency`. This however comes with
+a significant challenge: the TypeScript compiler API is not stable between
+TypeScript releases, and it is hence not possible to author a `jsii` compiler
+that can leverage an arbitrary TypeScript version.
+
+#### Only de-couple the `jsii` package version, but stick with [SemVer]
+
+It would technically be possible to de-couple the `jsii` package from the rest
+of the toolchain's versioning, and to issue a new `jsii` major version each time
+we update the [TypeScript] compiler it internally uses. This would however make
+it difficult for customers to understand the relationship between `jsii` major
+releases and the [TypeScript] language level they support.
+
+Using the same version number prefix (`major.minor`) as the [TypeScript]
+compiler makes the relationship clear and removes the need to maintain separate
+documentation for customers to understand what they are upgrading to.
+
+### What are the drawbacks of this solution?
+
+This solution is a stark departure from versioning schemes and maintenance
+policies used by AWS on open-source products. It implies offering critical bug
+fixes and security updates for up to 5 different release lines (1 current, and
+3 to 4 previous), which might require significant effort.
+
+Following the [TypeScript] `major.minor` reduces our ability to introduce
+breaking changes to the `jsii` compiler, as these must be timed together with
```

Comment:
What kind of breaking changes would we be intending to release? Anything more than removal of deprecated features?

I think I would still feel better if we chose not to introduce any breaking changes on top of the potential syntactic breaking changes introduced by the underlying TypeScript (but I'm fine with deprecation+removal a good while later).